ext2/ext2.c

```diff
@@ -3354,6 +3354,12 @@ int ext2_open(
         ERAISE(-ENOTDIR);
     }
 
+    if ((flags & O_TMPFILE) && ((flags & O_RDWR) || (flags & O_WRONLY)) &&
+        S_ISDIR(locals->inode.i_mode))
+    {
+        ERAISE(-EISDIR);
```

Comment:
Are you fixing https://github.com/deislabs/mystikos/issues/513? Can you do the same for ramfs and hostfs?

Reply:
I have added similar check in ramfs. As hostfs can actually support tmpfiles, I didn't make any change there.